atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/expectations/TrackingIterator.java

```diff
@@ -40,7 +43,13 @@ protected I delegate() {
     @Override
     public T next() {
         T result = delegate.next();
-        tracker.accept(measurer.applyAsLong(result));
+
+        try {
+            tracker.record(measurer.applyAsLong(result));
+        } catch (Exception exception) {
+            log.warn("Data tracking failed", exception);
```

Comment:
nit: Consider adding more detail as to what the implications are. Something like `Failed to track data. This means that reported data will be inaccurate, which may hide some transaction badness.`